index.js

```diff
@@ -59,14 +60,19 @@ module.exports = fp(async function (app, opts) {
 
   const root = {}
   let schema = opts.schema
+  let gateway = opts.gateway
+
+  if (gateway && (schema || opts.resolvers || opts.loaders || opts.subscription)) {
```

Comment:
Thanks for this PR and the federation implementation 😃👍 

Why subscriptions aren't available in a federated environment ? Wouldn't it be possible to forward the websocket connection to the related services ?

Reply:
Subscription support is coming but only after this first version is working as expected. This PR already adds a lot of complexity, and I would like to keep the PR as small as possible.

Reply:
Nice ! I was wondering if there was any limitation, thanks !

Reply:
To be honest I haven't looked into it yet. But I don't think it is as simple as forwarding WebSocket connections to the related services because you can fetch data in the subscription service that lives in some other service.

Reply:
I don't know much about federation, but for what I know about subscriptions in apollo-server:

A subscription is defined with two functions a `subscribe` and `resolve`, subscribe could be forwarded to the service which declared the subscription and I think we could resolve the subscription the same way the gateway retrieves the fragments for a federated query or mutation. But I don't know if this would work or cover every use cases 🤔.